Forge.TreeWalker.UnitTests/test/TreeWalkerUnitTests.cs

```diff
@@ -500,12 +500,71 @@ public void TestTreeWalkerSession_WalkTree_Failed_MissingKey()
         public void TestTreeWalkerSession_WalkTree_NoChildMatched()
         {
             this.TestInitialize(jsonSchema: ForgeSchemaHelper.NoChildMatch);
-
             // Test - WalkTree and expect the Status to be NoChildMatched.
             string actualStatus = this.session.WalkTree("Root").GetAwaiter().GetResult();
             Assert.AreEqual("RanToCompletion_NoChildMatched", actualStatus, "Expected WalkTree to end with NoChildMatched status.");
         }
 
+        [TestMethod]
+        public void TestReexecutingNode_WithRetryCurrentTreeNodeActionsFlag_Success()
+        {
+            this.TestInitialize(jsonSchema: ForgeSchemaHelper.ReExecuteNodeSchema);
+
+            this.session.WalkTree("Root").GetAwaiter().GetResult();
+            string time1 = session.GetLastActionResponse().Status;
+
+            this.session = new TreeWalkerSession(new TreeWalkerParameters(
+                this.sessionId,
+                parameters.ForgeTree,
+                parameters.ForgeState,
+                this.callbacksV2,
+                this.token)
+            {
+                UserContext = new ForgeUserContext(),
+                ForgeActionsAssembly = typeof(CollectDiagnosticsAction).Assembly,
+                InitializeSubroutineTree = this.InitializeSubroutineTree,
+                Dependencies = new List<Type>() { typeof(FooEnum) },
+                ScriptCache = this.scriptCache
+            });
+
+            this.session.WalkTree("Root").GetAwaiter().GetResult();
+            string time2 = session.GetLastActionResponse().Status;
+
+            Assert.IsTrue(time1 == time2);
+
+        }
+
+        [TestMethod]
+        public void TestReexecutingNode_WithoutRetryCurrentTreeNodeActionsFlag_Success()
+        {
+            this.TestInitialize(jsonSchema: ForgeSchemaHelper.ReExecuteNodeSchema);
+            this.session.Parameters.RetryCurrentTreeNodeActions = true;
+
+            this.session.WalkTree("Root").GetAwaiter().GetResult();
+            string time1 = session.GetLastActionResponse().Status;
+
+            this.session = new TreeWalkerSession(new TreeWalkerParameters(
+                this.sessionId,
+                parameters.ForgeTree,
+                parameters.ForgeState,
+                this.callbacksV2,
+                this.token)
+            {
+                UserContext = new ForgeUserContext(),
+                ForgeActionsAssembly = typeof(CollectDiagnosticsAction).Assembly,
+                InitializeSubroutineTree = this.InitializeSubroutineTree,
+                Dependencies = new List<Type>() { typeof(FooEnum) },
+                ScriptCache = this.scriptCache,
+                RetryCurrentTreeNodeActions = true
+            });
+
+            this.session.WalkTree("Root").GetAwaiter().GetResult();
+            string time2 = session.GetLastActionResponse().Status;
+
+            Assert.IsFalse(time1 == time2);
```

Comment:
>Assert.IsFalse(time1 == time2); [](http://example.com/codeflow?start=12&length=31)

Instead of adding a new test Action, could we re-use the RevisitAction? I'm always a bit cautious about relying on DateTime.UtcNow in UTs because they can execute so quickly and create weirdness.

RevisitAction works like this. Every time it is executed, it looks at the PreviousActionResponse and increases it's ActionResponse.Output by 1.

In the first UT (without the new flag set), we would expect both outputs to be 0 (because the Action doesn't get re-executed).

In the second UT (with the new flag set), we would expect the first output to be 0 and the retry to be 1 (because the Action was re-executed).